### ISSUE TITLE
Decimal mul returns a Uint128

### ISSUE THREAD

**Issue by archseer** [CosmWasm/cw-plus#521]
The comment says 
```
/// Both d*u and u*d with d: Decimal and u: Uint128 returns an Uint128. There is no
/// specific reason for this decision other than the initial use cases we have. If you
/// need a Decimal result for the same calculation, use Decimal(d*u) or Decimal(u*d).
```

but the constructor `Decimal(_)` is private. 

There is also no way to construct a decimal statically, let's say if I wanted to I pre-calculate some constants at compile time and I already pre-scaled the value correctly, i.e.

```rust
pub const fn decimal(i: usize, decimals: u8) -> u128 {
    let decimals = 10u128.pow(decimals as u32);
    i as u128 * decimals
}

const MY_NUM: Decimal = Decimal(decimal(10, 18));
```

**Comment by archseer** [comment]
Looks like the first part was fixed in `1.0.0` series.

**Comment by archseer** [comment]
(Opened this on the wrong repo)
